### PR TITLE
4bpp texture palette support + modified paletted_pcx sample through the use of USE_16C_PALETTE

### PR DESCRIPTION
**Pull request by ozzyyzzo4096**
Paletted textures are still represented internally as 32b (GL_RGBA8) , 16b formats would be a nice addition later on , for more resident palettes at once.

### CODE REVIEW

File: GL/texture.c [hunk 1]
```diff
@@ -139,6 +139,41 @@ static void GPUTextureTwiddle8PPP(void* src, void* dst, uint32_t w, uint32_t h)
...
+	uint8 * pixels;
+    uint16 * vtex;
+    pixels = (uint8 *) src;
+    vtex = (uint16*)dst;
+
+    for (y=0; y<h; y += 2) {
+        yout = y;
+        for (x=0; x<w; x += 2) {
+
+                //Ozzy note: if x alignment is wrong just use version commented instead (possible error)
```
Comment (Kazade) [inline GL/texture.c:157]: Any more information on this? I think we need to resolve this (and remove the #if) before we merge. Is there an edge case bug?

File: GL/texture.c [hunk 2]
```diff
@@ -1487,7 +1535,7 @@ GLAPI void APIENTRY glColorTableEXT(GLenum target, GLenum internalFormat, GLsize
     }
 
     palette->data = (GLubyte*) malloc(width * 4);
-    palette->format = format;
+    palette->format = format; //Ozzy:was previously forcing to GL_RGBA for testing.
```
Comment (Kazade) [inline GL/texture.c:1538]: Probably can remove comment?

File: GL/texture.c (hunk 1 above)
Comment (ozzyyzzo4096) [reply GL/texture.c:157]: well, the usual 4bpp madness with interlaced pixels [0,1] or [1,0] ..  maybe i've done it in the wrong order ? not sure about it ^^ 

make your choice :)

File: GL/texture.c (hunk 2 above)
Comment (ozzyyzzo4096) [reply GL/texture.c:1538]: yup.

**Pull request merged**